### PR TITLE
Added import of multiple text files; also ensured that Excel files are consistently imported correctly

### LINKED ISSUE #6476: Reading of multiple txt files
I am trying to read (climatic) data of course.  There are multiple files per element, with one file per year.  In a file that is shared the tmax has missing as -100 and reading an individual file it works fine reading 31 rows.  There are 13 columns with the day of the month plus one for each month:
image

These are txt files and in the import dialogue I use the fixed width option as shown below:
image

When I switch to read the multiple files is changes to csv as shown below and doesn't read the 31 rows.  It reads 28 usually and 29 in a leap year.

image

### LINKED ISSUE #6482: Small bug in importing Excel file
This is an awkward file - sent separately to @Patowhiz .  The bug may be more general and is (I think) easy to fix.
I am importing a selection from the file as shown:
image

(So no column names, skipping 281 rows and importing 75 rows.)

The commands generated were as follows:
```
# Code generated by the dialog, Import Dataset
vtary_muchinga <- rio::import(file="C:/Users/RogerStern/Dropbox (SSD)/Roger/Climatic/Zambia/Voluntary/Voluntary stations.xlsx", guess_max=Inf, which=1, skip=281, col_names=FALSE, n_max=74)
data_book$import_data(data_tables=list(vtary_muchinga=vtary_muchinga))

rm(vtary_muchinga) 
```
But I got it wrong first time and only imported 74 rows.  So I deleted the data file and tried again.  Then it went wrong and imported each variablew into a different data frame:

image

The commands were now:

```
# Code generated by the dialog, Import Dataset
vtary_muchinga <- rio::import(file="C:/Users/RogerStern/Dropbox (SSD)/Roger/Climatic/Zambia/Voluntary/Voluntary stations.xlsx", guess_max=Inf, which=1, skip=281, col_names=FALSE, n_max=75)
data_book$import_data(data_tables=vtary_muchinga)
```
You can see that the main rio:import command is the same, but the `data_book$import` line has changed.  I assume that is the problem?

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes #6482 and Fixes #6476
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

**Review by lloyddewit — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern please can you test? thanks

**Comment by Patowhiz** [comment]
@rdstern you can now also test if this PR fixes issue #6476. I had to push changes here because they affect the same dialog. Thanks.

**Comment by Patowhiz** [comment]
@lloyddewit I think this should now also have a "feature" label.

**Comment by rdstern** [comment]
@Patowhiz great.  This now works fine on the import, with just the extension given as the sheet name.
But I wonder what should happen next?  When I open again it seems ok as well.

However, I tried closing the Data Book to go back to your introductory screen.  I think then perhaps the Open dialogue should reset?    (I suggest this is a reasonable thing to do, if the import of multiple files didn't work as well as would be hoped - that's the situation with the Pakistan files, where we find they are inconsistent.)

Currently, when I then try to open an rds file I get:

image

If I try to open a similar single file to before I get:

image

In each case I have to press Reset.  Then I get the default directory:
image

Aha.  I need to press REset on the dialogue.  Then I close it and the try again from the start-up screen.  

I wonder what should happen?

**Comment by lloyddewit** [comment]
> 
> 
> @lloyddewit I think this should now also have a "feature" label.

agreed

**Comment by Patowhiz** [comment]
@rdstern thanks for the multiple testing. That's definitely a bug probably cause by my new additions. Let me fix that.

**Comment by Patowhiz** [comment]
@rdstern I'm unable to reproduce the bug you described above. Closed the databook as explained above and reopened the dialog and it still retained its previous state. Perhaps a skype meeting to demonstrate how you got this bug or steps that I can take to reproduce it

**Comment by Patowhiz** [comment]
@rdstern I have fixed the bug you described above. You can now review.

**Review by rdstern — APPROVED** [review]
From my point of view this seems ok

**Comment by Patowhiz** [comment]
This PR is blocking issue #6196 and refactoring of the import dialog. It may potentially also block issue #6556.

File: instat/dlgImportDataset.vb [hunk 1]
```diff
@@ -959,9 +1001,9 @@ Public Class dlgImportDataset
     End Sub
 
     Private Sub ucrBase_ClickOk(sender As Object, e As EventArgs) Handles ucrBase.ClickOk
-        ' add the item to the MRU (Most Recently Used) list...
-        'only add if its a file that was selected
-        If Not ucrChkMultipleFiles.Checked Then
+        'add the item to the MRU (Most Recently Used) list...
+        'only add if its a file that was selected, don't folder paths
```
Comment (lloyddewit) [inline instat/dlgImportDataset.vb:1005]:
```suggestion
        'only add if its a file that was selected, don't add if folder path was selected
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern I made a small change to one of the comments. If you can approve again, then I'll merge, thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**